mono/metadata/domain.c

```diff
@@ -1888,26 +1888,26 @@ get_runtimes_from_exe (const char *file, MonoImage **out_image, const MonoRuntim
 		 * If there are no such elements, look for a requiredRuntime element.
 		 */
 		if (app_config->supported_runtimes != NULL) {
-			int n = 0;
 			GSList *list = app_config->supported_runtimes;
 			while (list != NULL) {
 				version = (char*) list->data;
 				runtime = get_runtime_by_version (version);
 				if (runtime != NULL)
-					runtimes [n++] = runtime;
+					runtimes = g_slist_prepend (runtimes, (gpointer)runtime);
 				list = g_slist_next (list);
 			}
-			runtimes [n] = NULL;
+			runtimes = g_slist_reverse (runtimes);
```

Comment:
why?

also: `runtimes` starts out NULL when `get_runtimes_from_exe` is first called, right?  But if it wasn't - this would reverse the input portion of the list too, right?  That seems... undesirable.

Reply:
I wanted to preserve the order from the XML, and I believe prepending and then reversing is faster than appending a bunch of times. You're right about reversing the input portion, though I also agree that there's no reason for it to take an input runtimes at all, so I'm inclined to leave this once I fix that.

Reply:
Sounds good to me.